packages/nimbus/src/components/money-input/money-input.tsx

```diff
@@ -283,38 +269,31 @@ export const MoneyInputComponent = (props: MoneyInputProps) => {
     [handleCurrencyChange]
   );
 
-  const stateProps = {
-    isInvalid,
-    isDisabled,
-    isReadOnly,
-    hasFocus,
-  };
+  const noCurrenciesLabelId = useId();
+  const highPrecisionBadgeId = useId();
 
   return (
-    <MoneyInputRootSlot
-      {...recipeProps}
-      {...styleProps}
-      {...stateProps}
-      {...remainingProps}
-    >
+    <MoneyInputRootSlot {...recipeProps} {...styleProps} {...remainingProps}>
       <MoneyInputContainerSlot>
         {/* Currency Select or Label */}
-        <MoneyInputCurrencySelectSlot {...stateProps}>
+        <MoneyInputCurrencySelectSlot>
           {hasNoCurrencies ? (
             <MoneyInputCurrencyLabelSlot asChild>
-              <label data-testid="currency-label">{value.currencyCode}</label>
+              <label id={noCurrenciesLabelId}>{value.currencyCode}</label>
             </MoneyInputCurrencyLabelSlot>
           ) : (
             <Select.Root
+              id={currencySelectId}
+              name={currencySelectName}
```

Comment:
If we're going to generate a name and Id for the currencySelect, we should make sure those are actually in the DOM (otherwise a lot of formik forms and tests on the MC side will prolly break)